azure/scope/cluster.go

```diff
@@ -25,7 +25,8 @@ import (
 	"strconv"
 	"strings"
 
-	asonetworkv1 "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
+	asonetworkv1api20201101 "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	asonetworkv1api20220701 "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
```

Comment:
why do we need 2 different network API versions?

Reply:
ASO only has subnet/vnet resources with 20201101 and nat gateways and bastion hosts with 20220701: https://azure.github.io/azure-service-operator/reference/network/.

Compared to AKS resources which have multiple API versions available: https://azure.github.io/azure-service-operator/reference/containerservice/